src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -118,16 +118,27 @@
                              :password     password
                              :on-completed on-transactions-modal-completed}})))
 
+(defn discard-transaction
+  [{{:wallet/keys [send-transaction] :as db} :db}]
+  (let [{:keys [transaction-id]} send-transaction]
+    (merge {:db (update db :wallet/send-transaction assoc
+                        :signing? false
+                        :transaction-id nil
+                        :wrong-password? false)}
+           (when transaction-id
+             {:discard-transaction transaction-id}))))
+
 (handlers/register-handler-fx
   :wallet/discard-transaction
-  (fn [{{:wallet/keys   [send-transaction] :as db} :db} _]
-    (let [{:keys [transaction-id]} send-transaction]
-      (merge {:db (update db :wallet/send-transaction assoc
-                          :signing? false
-                          :transaction-id nil
-                          :wrong-password? false)}
-             (when transaction-id
-               {:discard-transaction transaction-id})))))
+  (fn [cofx _]
+    (discard-transaction cofx)))
+
+(handlers/register-handler-fx
+  :wallet/discard-transaction-leave
```

Comment:
tbh i don't like the idea to hide navigation into event, it's better if we can see it in view, as was before

Reply:
and if we want this it should be named like `discard-transaction-and-navigate-back`, but still i preferer to dispatch 2 events from view, its more clear